### DISCUSSION TITLE
Authentication for proxy domain

### DISCUSSION THREAD

**Discussion by jtp03a** [coder/code-server#3240] [Q&A] [answered]
Is there anyway to disable authentication for web services that are being exposed via the --proxy-domain flag? For instance I have a react app running in code-server on port 3000 and accessible via 3000.codeserver.mydomain.com. I also have an express backend running on port 4000 and accessible via 4000.codeserver.mydomain.com. However when I try and send a axios request from my frontend to my backend I am getting an authentication error. I understand that in a production deployment ill be proxying my frontend to my backend and all running in a single container and not concerned about cors issues but for testing purposes I would like to be able send requests to my api at those subdomains.

**Comment by jsjoeio** [comment, marked as answer]
Hmm..not that I know of but maybe @code-asher knows. 

> express backend

Could you disable CORS on your Express server in your dev environment? According to [the docs](https://expressjs.com/en/resources/middleware/cors.html), I think you could set `origin: false` maybe?

**Reply by code-asher** [reply]
No way at the moment, tracking here: https://github.com/cdr/code-server/issues/1944

**Reply by jtp03a** [reply]
Copy all, thanks for the reply. In the interim does anyone know what is the best way to authenticate with code-server in an axios request? I know that if I am using postman to test the API I can setup a cookie to send along with the key=sha256_of_code_server_password and that works, but I'm not sure how to force that in an axios request.

**Reply by code-asher** [reply]
You might need to add `withCredentials: true` to make it include cookies? https://stackoverflow.com/questions/43002444/make-axios-send-cookies-in-its-requests-automatically
